wremnants/CardTool.py

```diff
@@ -281,7 +281,13 @@ def addSystematic(self, name, systAxes=[], outNames=None, skipEntries=None, labe
 
         if name == self.nominalName:
             logger.debug(f"Defining syst {rename} from nominal histogram")
-            
+        
+        # precompile splitGroup expressions for better performance
+        splitGroupDict = {g: re.compile(v) for g, v in splitGroup.items()}
+        # add the group with everything if not there already
+        if group not in splitGroupDict:
+            splitGroupDict[group] = re.compile(".*")
```

Comment:
ok, perhaps this is included back here then (I'll keep the previous comment in case I was missing something else)